WooCommerce/Classes/Authentication/Navigation Exceptions/JetpackConnectionWebViewModel.swift

```diff
@@ -10,30 +10,34 @@ final class JetpackConnectionWebViewModel: AuthenticatedWebViewModel {
     let initialURL: URL?
     let siteURL: String
     let completionHandler: () -> Void
+    let failureHandler: () -> Void
     let dismissalHandler: () -> Void
 
     private let stores: StoresManager
     private let analytics: Analytics
     private var isCompleted = false
+    private var isFailure = false
 
     init(initialURL: URL,
          siteURL: String,
          title: String = Localization.title,
          stores: StoresManager = ServiceLocator.stores,
          analytics: Analytics = ServiceLocator.analytics,
          completion: @escaping () -> Void,
+         onFailure: @escaping () -> Void = {},
```

Comment:
Just a 💭

Should we try to clarify that `onFailure` is fired when we encounter 404? I am asking this because we specifically treat this as 404 [here](https://github.com/woocommerce/woocommerce-ios/pull/9354/files#diff-5b2d3d7f53b79702e24015df8d058f2c45f0b6ba5e79eba6db48da378af75cccR134-R139)

Reply:
Good point! I updated the logic to send the error code from the web view model and display it on the error screen as is in 5c2ba687415833a384790c0b650ce79e35b87e32.